doc/dev/index.rst

```diff
@@ -58,3 +58,56 @@ A registry is an in-memory data-store which is backed by a repository.
 
 Job monitoring
 --------------
+
+IGangaFile
+----------
+
+All file types as of Ganga 6.1 inherit from ```IGangaFile```. This main exception to this is the ```File``` object which as of 05/05/2016 is used as it still has more features than the ```IGangaFile``` inheirted classes do.
```

Comment:
Rats, wasn't planning to add this to this PR... oh well, I'll just modify it, no harm in extra docs